src/edu/cmu/tartan/Game.java

```diff
@@ -564,9 +611,200 @@ private void help() {
 		PrintMessage.printConsole("- " + indirobj.toString() + "\n");
 		PrintMessage.printConsole("- " + misc.toString() + "\n");
 		PrintMessage.printConsole("- You can inspect an inspectable item by typing \"Inspect <item>\"\n");
+		PrintMessage.printConsole("- You can save the game by typing \"save\"\n");
+		PrintMessage.printConsole("- You can periodic save the game by typing \"save <minute>\"\n");
+		PrintMessage.printConsole("- You can load the game by typing \"load\"\n");
 		PrintMessage.printConsole("- You can quit by typing \"quit\"\n");
 		PrintMessage.printConsole("- Good luck!\n");
 
 	}
 
+	public JSONObject roomArrayListConvertJSONObject(){
+        //save game's room list
+        JSONObject roomList = new JSONObject();
+
+        for(Room r : roomArrayList){
+            JSONObject room = new JSONObject();
+            JSONArray itemList = new JSONArray();
+            for(Item i : r.items){
+                itemList.add(i.toString());
+            }
+            room.put("room",r.shortDescription());
+            room.put("items",itemList);
+            roomList.put("room"+roomArrayList.indexOf(r),room);
+        }
+        return roomList;
+    }
+
+    public JSONArray visitedRoomListConvertJSONArray(){
+        //save vistied room list
+        JSONArray visitedRooms = new JSONArray();
+        for(Room r : player.getRoomsVisited()){
+            visitedRooms.add(r.shortDescription());
+        }
+        return visitedRooms;
+    }
+
+    public JSONArray collectedItemListConvertJSONArray(){
+        //save player's item list
+        JSONArray collectedItems = new JSONArray();
+        for(Item i : player.getCollectedItems()){
+            collectedItems.add(i.toString());
+        }
+        return collectedItems;
+    }
+
+    public void save(int period){
+		SaveTask saveTask = new SaveTask(this);
+		saveScheduler.schedule(saveTask,100,period);
```

Comment:
To stop scheduler when a program stops, please check below site.
https://code.i-harness.com/ko/q/f8c7